specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/workbooks_API.json

```diff
@@ -0,0 +1,751 @@
+            "schema": {
+              "$ref": "#/definitions/WorkbookErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "WorkbookUpdate": {
+            "$ref": "./examples/WorkbookUpdate.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/microsoft.insights/workbooks/{resourceName}": {  
+      "get": {
+        "description": "Get a single workbook by its resourceName.",
+        "operationId": "Workbooks_Get",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkbookResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "A workbook definition.",
+            "schema": {
+                "$ref": "#/definitions/Workbook"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/WorkbookErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "WorkbookGet": {
+            "$ref": "./examples/WorkbookGet.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/microsoft.insights/workbooks/links": {  
+      "get": {
+        "description": "Gets a list of workbooks.",
+        "operationId": "Workbooks_List",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/SourceIdParameter"
+          },
+          {
+            "$ref": "#/parameters/CategoryParameter"
+          },
+          {
+            "$ref": "#/parameters/TagsParameter"
+          },
+          {
+            "$ref": "#/parameters/CanFetchWorkbookContentParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "A list containing 0 or more workbook definitions.",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/Workbook"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/WorkbookErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "WorkbooksListLink": {
+            "$ref": "./examples/WorkbooksListLink.json"
+          }
+        }
+      },    
+      "put": {
```

Comment:
This PUT URL should be modeled as /workbooks/{workbookname}/links/{linkName}. Then in the request body you dont need the targetId parameter as it can be inferred from the URL.